src/proxy/ClientTable.php

```diff
@@ -107,65 +107,108 @@ public function getOffsetTotal()
     
     public function isComplet()
     {
-        return $this->getRows() == count($this->getContentRows());
+        return $this->getRows() == $this->_contentRowsCount;
     }
     
-    private $_contentRows;
-    
     public function getContentRows()
     {
-        if ($this->_contentRows === null) {
-            $data = [];
-            
-            Console::startProgress(0, $this->getOffsetTotal(), 'Fetch: ' . $this->getName() . ' ');
-            
-            $progress = 1;
-            for ($i=0; $i<$this->getOffsetTotal(); $i++) {
-                $requestData = $this->request($i);
-                
-                if (!$requestData) {
-                    continue;
-                }
-                
-                $data = array_merge($requestData, $data);
-                gc_collect_cycles();
-                Console::updateProgress($progress++, $this->getOffsetTotal());
-            }
-            
-            Console::endProgress();
-            $this->_contentRows = $data;
-        }
-        
-        return $this->_contentRows;
+	    throw new \ErrorException('Not implemented');
     }
-    
-    public function syncData()
-    {
-        Yii::$app->db->createCommand()->truncateTable($this->getName())->execute();
-        
-        $rows = $this->getContentRows();
 
-        $chunks = array_chunk($rows, 25);
-        
-        foreach ($chunks as $match) {
-            Yii::$app->db->createCommand()->batchInsert($this->getName(), $this->cleanUpBatchInsertFields($this->getFields()), $this->cleanUpMatchRow($match))->execute();
-        }
+    private $_contentRowsCount;
+
+    public function getContentRowCount()
+    {
+    	return $this->_contentRowsCount;
     }
     
-    private function request($offset)
+    public function syncData()
     {
-        $curl = new Curl();
-        $curl->get($this->build->requestUrl, ['machine' => $this->build->machineIdentifier, 'buildToken' => $this->build->buildToken, 'table' => $this->name, 'offset' => $offset]);
-        
-        if (!$curl->error) {
-            $response = Json::decode($curl->response);
-            $curl->close();
-            unset($curl);
-            return $response;
-        } else {
-            $this->build->command->outputError("Error while collecting data from server: " . $curl->error_message);
-        }
-        
-        return false;
+	    try {
```

Comment:
I don't understand the reason why to pack into try / finally block? Its not wrong and i am not against, just wondering what is the intention behind.

Reply:
If something goes wrong after the SQL_MODE has changed I'm not sure it will makes other problems. With the try-block I will be safety that the SQL_MODE is set to the old value. Same thing on with FOREIGN_KEY_CHECKS and UNIQUE_CHECKS.

Reply:
But if something happens wrong while set SQL mode, the sync block wont executed anymore:

``` 
Yii::$app->db->createCommand()->truncateTable($this->getName())->execute();
$this->syncDataInternal();
```

So this part of code should always be executed whether sql_mode set works or not. Actually i am not very experienced with `'SET SESSION UNIQUE_CHECKS = 0` querys and i am not sure if there are any permission required on the sql server in order to work (or any other problems).

Reply:
To change the session system variables FOREIGN_KEY_CHECKS, UNIQUE_CHECKS and SQL_MODE require no special privileges. There should be nothing go wrong and `syncDataInternal` will be execute anyway. 

`SET UNIQUE_CHECKS` is just the short version of `SET SESSION UNIQUE_CHECKS`. I am with you without the SESSION modifier it is more clearly.

I did some refactoring of this code yet.